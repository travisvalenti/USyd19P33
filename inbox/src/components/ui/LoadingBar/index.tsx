import React from 'react'

const LoadingBar: React.FC = () => (
  <div className="LoadingBar">
    <div></div>
  </div>
)

export default LoadingBar
