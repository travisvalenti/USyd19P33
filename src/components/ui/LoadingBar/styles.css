.LoadingBar {
  width: 100%;
  height: 10px;
  background-color: white;
  margin: 0.5em 0;
  overflow: hidden;
  border-radius: 5px;
  box-sizing: border-box;
}

.LoadingBar > div {
  height: 100%;
  border-radius: 5px;
  background-color: dodgerblue;
  animation: load 1.8s infinite ease-in-out;
}

@keyframes load {
  0% {
    width: 0%;
    margin-left: 0%;
  }

  50% {
    width: 100%;
    margin-left: 0%;
  }

  100% {
    width: 0%;
    margin-left: 100%;
  }
}
