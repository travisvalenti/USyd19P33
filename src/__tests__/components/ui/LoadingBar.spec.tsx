import React from 'react'
import { render } from '@testing-library/react'

import LoadingBar from '../../../components/ui/LoadingBar'

describe('LoadingBar component', () => {
  test('Matches the snapshot', () => {
    const loadingBar = render(<LoadingBar />)
    expect(loadingBar.container).toMatchSnapshot()
  })
})
