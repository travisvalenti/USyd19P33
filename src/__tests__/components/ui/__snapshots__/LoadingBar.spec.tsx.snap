// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingBar component Matches the snapshot 1`] = `
<div>
  <div
    class="LoadingBar"
  >
    <div />
  </div>
</div>
`;
